packages/taquito-signer/src/derivation-tools/ecdsa.ts

```diff
@@ -0,0 +1,136 @@
+    /**
+     * @param seedSrc result of Bip39.mnemonicToSeed
+     * @param curve known supported curve p256 or secp256k1
+     * @returns instance of PrivateKey non-HD keys derived
+     */
+    static fromSeed(seedSrc: Uint8Array | string, curve: CurveName): PrivateKey {
+        let seed = typeof seedSrc === "string" ? parseHex(seedSrc) : seedSrc;
+        if (seed.length < minSeedSize || seed.length > maxSeedSize) {
+            throw new InvalidSeedLengthError(seed.length);
+        }
+        if (!Object.prototype.hasOwnProperty.call(seedKey, curve)) {
+            throw new InvalidCurveError(`unknown curve ${curve}`);
+        }
+        const c = new ec(curve);
+        if (c.n?.bitLength() !== 256) {
+            throw new InvalidBitSize(`invalid curve bit size ${c.n?.bitLength()}`);
+        }
+
+        const key = new TextEncoder().encode(seedKey[curve]);
+        let d: BN | null = null;
+        let chain: Uint8Array = new Uint8Array();
+        let i = 0;
+        while (i === 0) {
+            const sum = new HMAC(SHA512, key).update(seed).digest();
+            d = new BN(sum.subarray(0, 32));
+            chain = sum.subarray(32);
+            if (d.isZero() || d.cmp(c.n as BN) >= 0) {
+                seed = sum;
+            } else {
+                i++
+            }
+        }
+
+        const keyPair = <KeyPair>c.keyPair({});
+        keyPair.priv = d;
+        return new PrivateKey(keyPair, chain);
+    }
+/**
+ *
+ * @param index derivation path item pre-harded if applicable ie: 44' -> 2^31 + 44
+ * @returns child PrivateKey of the current PrivateKey
+ */
+    derive(index: number): PrivateKey {
+        const data = new Uint8Array(37);
+        if ((index & Hard) !== 0) {
+            // hardened derivation
+            data.set(this.keyPair.getPrivate().toArray(), 1);
+        } else {
+            data.set(this.keyPair.getPublic().encodeCompressed(), 0);
+        }
+        new DataView(data.buffer).setUint32(33, index);
+
+        let d: BN = new BN(0);
+        let chain: Uint8Array = new Uint8Array;
+        let i = 0;
+        while (i === 0) {
+            const sum = new HMAC(SHA512, this.chainCode).update(data).digest();
+            d = new BN(sum.subarray(0, 32));
+            chain = sum.subarray(32);
+            if (this.keyPair.ec.n && d.cmp(this.keyPair.ec.n as BN) < 0) {
+                d = d.add(this.keyPair.getPrivate() as BN).mod(this.keyPair.ec.n as BN);
+                if (!d.isZero()) {
+                    i++
+                }
+            }
+            data.set(chain, 1);
+            data[0] = 1;
+        }
+        const keyPair = <KeyPair>this.keyPair.ec.keyPair({});
+        keyPair.priv = d;
+        return new PrivateKey(keyPair, chain);
+    }
+/**
+ *
+ * @param path pre-hardened (if applicable) derivation path items ie 44'/1729'/0/0 -> 2^31 + 44/2^31 + 1729/0/0
+ * @returns final child of the full HD keys derivation
+ */
+    derivePath(path: Iterable<number>): PrivateKey {
```

Comment:
Do we need this method, it doesn't appear to be used.

Reply:
i am not sure why i want using is... i just be default did it myself in the helper function

ecdsa is switched to use the method 

however interestingly ed25519 comes out with the incorrect key so i'm not using that one...